### DISCUSSION TITLE
采用基于jenkins + jar部署，执行sshPublisher报错

### DISCUSSION THREAD

**Discussion by hellon** [zuihou/zuihou-admin-cloud#271] [01.常见问题]
根据文档进行jenkins配置完成后，进行jenkins构建部署，到“推送jar&执行动作”这一步出现错误。
错误截图：
![image](https://user-images.githubusercontent.com/15033782/231081306-899ed82c-6c2a-4f54-8076-ab46bc22c10f.png)
我查看Jenkinsfile文件发现，需要执行`server_run.sh`这个脚本，如下，这个文件需要我们手动传到jenkins服务器上吗，还是会自动创建。

`stages {
        stage('替换环境参数') {
            steps {
                script {

                    // SERVER_SSH_HOST一定要包含_ 如：ip_dev ip_prod 等格式
                    PROFILES_ARR = "${SERVER_SSH_HOST}".split('_')
                    PROFILES = PROFILES_ARR[1]

                    // 工作空间
                    WORKSPACE_HOME = "/data_${PROFILES}"
                    // 服务端执行的脚本
                    EXEC_COMMAND = "bash -x -s < ${WORKSPACE_HOME}/bin/server_run.sh ${JAR_NAME} ${PROFILES} ${ACTION}"
                    echo "您选择了如下参数："
                    echo "拉取分支： ${branch}, 打包命令：${MAVEN_COMMAND}，是否推送jar：${IS_PUSH_JAR}, 将${SERVER_NAME} 推送至：${SERVER_SSH_HOST}, 运行环境参数：${PROFILES}，启动动作：${ACTION}"
                }
            }
        }`

**Comment by zuihou** [comment]
手动上传

**Reply by zuihou** [reply]
server_run.sh 在gitlab  docs 项目里

**Comment by hellon** [comment]
server_run.sh文件是放到jenkins服务器上还是放到要部署服务的服务器上，我放到了jenkins服务器上还是出现了上次的错误
`ERROR: Exception when publishing, exception message [Exec exit status not zero. Status [1]]`

**Reply by zuihou** [reply]
部署服务器

**Reply by zuihou** [reply]
EXEC_COMMAND = "bash -x -s < ${WORKSPACE_HOME}/bin/server_run.sh ${JAR_NAME} ${PROFILES} ${ACTION}"

这行命令是在 部署服务器上执行的。
